### ISSUE TITLE
Gitea Flatpak on Flathub

### ISSUE THREAD

**Issue by aplatypus** [go-gitea/gitea#25506]
### Feature Description

I recently installed using `Jellyfin` server and `Jellyfin` client as a `Flatpak`-s.  The experience feels nice and clean, Jellyfin picked-up my existing media files for example and 'saw' my devices.

In a similar light I'd like to see a `Gitea` server on [Flathub]().  

To me that just seems the user friendly way to offer service bases applications.  I understand that people want to play with Docker or containers, etc  and even install the service as a sys admin.   I'm a "user" -- I want a local version management capacity; it seems that I need to learn sysadmin 101 and 201 and ...

I have seen some earlier suggestions on the same idea for Flaphub installation for Gitea -- They seem to get closed with no  outcome.  If someone knows enough to instruct me on what to do, I don't mind making an initial PR ... I'm just a use case really.  For me a ready to use Flatpak solution  would be _Just GREAT!_  . . .

Hope enough people express the same desire -- For you guys to see how popular your project will become once it is a few keystrokes away rather than a sysadmin workshop ... Please consider.

Kind regards,
    aplatypus

**Comment by techknowlogick** [comment]
Hey, thanks for this ticket. I package Gitea through several channels, and have attempted Flatpak in the past but was limited by a restriction placed on the amount of RAM needed to build the binary. Last I investigated I believe I found that was a hard/unchangeable limit, not sure if that was indeed the case or has at least changed.

I would love to package for Flatpak, and if anyone has any advice/could offer assistance, I would be grateful.

**Comment by JakobDev** [comment]
I have a lot of experience with building Flatpaks, so I can do that.

But I think this is not a good idea. While Docker is a container format designed for Server Software, Flatpak is a container Format designed for Desktop Applications. So I don't know, if Gitea fully works in as a Flatpak. And it might need a few Hacks to do so. 

Why do we even need this? If you just want to use Gitea e.g. on your local Pi for your Personal Projects, just download the Binary and use it. It is easy to install. Just execute it, open the Install Page in your browser, select SQLite as Database (for a Gitea Instance that is only used by you, you don't need a "real" database), chose a Username and Password and everything is ready. If you want to connect Gitea to the Internet and use it for more, you should learn how to maintain a Server.

> in the past but was limited by a restriction placed on the amount of RAM needed to build the binary

That sounds weird, considering things like Chromium are build from source as Flatpak. I never heard about that before. 

The bigger Problem is, that at the Moment you can't upload a Flatpak to Flathub (only a few people are allowed to do this). This will change in the Future, but at the Moment, you will need to give Flathub your manifest and the Buildbot of Flathub builds the Flatpak for you. The big Problem here, that you don't have Internet access during the Build, so I think it's easier to just pack the prebuild Binary.

**Comment by techknowlogick** [comment]
> That sounds weird, considering things like Chromium are build from source as Flatpak. I never heard about that before.

Oh weird. Yeah, I didn't look into it much, so it very well could've just been something local to my machine at the time.

**Comment by aplatypus** [comment]
... speaking as a 'user' person; I would want a "Desktop container" or format that lets me run (for want of a better word) "**Solutions**".  When I saw the **Jellyfin** flatpak I felt much better for not needing to learn a whole bunch of stuff (about containers) just _so my daughter run movies on her tablet_. 

I suppose this is going to sound odd or perhaps heritical that someone wants a platform that functions for my/our use cases.  

As far as I know, "containers" _per se_ do not exist.  They are a construct of underlying facilities like namespaces, network space, etc. ; I don't see anything _sacred_ about how one balances isolation or sharing.  I do love the way we have the **Flatseal** tool to manage those resource/permission areas.   

Returning to the '_user type person_' perspective I saw something like "Gitea" as a versioning system appliance.  I might frame/perceive Jellyfin as a media utility.  Someone pointed-out that a well configured Flatpak is offering some value about what is 'good practice' with a service or application.  That makes sense.  Whereas holding-out that expertise to detour people to a garage server project with OCI containers and services ... _Hmmmm_  Yes, that makes a good learning experience, is it a "solution"?

It is a bit frustrating.  People have asked me about "marketing Linux" or even just technology.  I used to point folk to the seminal "Marketing Myopia" paper and I believe very few have gone from intellect to: "grok".   They may even get the academic or theoretical intent; and yet we see a 1940-s toaster sell as a household staple.  Heck, I use the same process to make toast on a campfire.  Making  toast is a "technology" ... Making (someone's) life better is invention.  It is very likely cave-peopl emade toast that way, too.  Morse code used is a technology; the telephone the messaging system "better" (_imho_).

Sorry for a 'rant' there, it seems like some want to advance the Linux space and deride bread and butter issues about what we need to BE to "_be the change we want_" (Ghandi)

**Comment by techknowlogick** [comment]
@aplatypus I completely agree. I (and others) package Gitea for several different platforms so that people can run it in the way they are most comfortable.

**Comment by JakobDev** [comment]
I have [created a Flatpak for Gitea](https://github.com/JakobDev/gitea-flatpak). It just repacks the Binary. The only included Database is SQLite, but if you have another Database already set up, you should be able to connect. You can take a look at it @techknowlogick .

As I already set, Flatpak is not designed for Server Software. The Flatpak Developers have said this multiple times. As long as you just use it for a few personal Projects on your PC, it should be good. But if you need a more complex setup, you will ran into issues.
